src/structures/buffers.rs

```diff
@@ -71,16 +71,15 @@ macro_rules! named_field_buffer_type {
             }
         }
 
-        impl TryFrom<$native_type> for $tss_type {
-            type Error = Error;
-
-            fn try_from(native: $native_type) -> Result<Self> {
+        impl From<$native_type> for $tss_type {
+            fn from(native: $native_type) -> Self {
                 let mut buffer = $tss_type {
                     size: native.0.len() as u16,
                     ..Default::default()
                 };
+                buffer.size = native.0.len() as u16;
```

Comment:
Isn't the `size` field assignment duplicated here?

Reply:
Yup, that is not how it is supposed to be. I'll send up PR in a sec.